sdk-java/src/main/java/io/littlehorse/sdk/worker/internal/LHServerConnectionManager.java

```diff
@@ -1,238 +1,44 @@
 package io.littlehorse.sdk.worker.internal;
 
-import com.google.common.base.Throwables;
-import io.grpc.stub.StreamObserver;
-import io.littlehorse.sdk.common.LHLibUtil;
 import io.littlehorse.sdk.common.config.LHConfig;
-import io.littlehorse.sdk.common.exception.InputVarSubstitutionError;
-import io.littlehorse.sdk.common.exception.LHSerdeError;
-import io.littlehorse.sdk.common.exception.LHTaskException;
-import io.littlehorse.sdk.common.proto.LHErrorType;
-import io.littlehorse.sdk.common.proto.LHHostInfo;
-import io.littlehorse.sdk.common.proto.LHTaskError;
 import io.littlehorse.sdk.common.proto.LittleHorseGrpc.LittleHorseStub;
-import io.littlehorse.sdk.common.proto.PollTaskResponse;
-import io.littlehorse.sdk.common.proto.RegisterTaskWorkerRequest;
-import io.littlehorse.sdk.common.proto.RegisterTaskWorkerResponse;
-import io.littlehorse.sdk.common.proto.ReportTaskRun;
-import io.littlehorse.sdk.common.proto.ScheduledTask;
 import io.littlehorse.sdk.common.proto.TaskDef;
-import io.littlehorse.sdk.common.proto.TaskStatus;
-import io.littlehorse.sdk.common.proto.VariableValue;
 import io.littlehorse.sdk.worker.LHTaskWorkerHealth;
-import io.littlehorse.sdk.worker.WorkerContext;
-import io.littlehorse.sdk.worker.internal.util.ReportTaskObserver;
 import io.littlehorse.sdk.worker.internal.util.VariableMapping;
-import java.io.Closeable;
-import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
-import java.util.ArrayList;
-import java.util.Date;
 import java.util.List;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
-import java.util.concurrent.Semaphore;
-import lombok.extern.slf4j.Slf4j;
 
-@Slf4j
-public class LHServerConnectionManager implements StreamObserver<RegisterTaskWorkerResponse>, Closeable {
-
-    public Object executable;
-    public Method taskMethod;
-    public LHConfig config;
-    public List<VariableMapping> mappings;
-    public TaskDef taskDef;
-
-    private List<LHServerConnection> runningConnections;
-    private LittleHorseStub bootstrapStub;
-    private ExecutorService threadPool;
-    private Semaphore workerSemaphore;
-    private Thread rebalanceThread;
-
-    private static final long HEARTBEAT_INTERVAL_MS = 5000L;
+public class LHServerConnectionManager {
 
+    private final RebalanceThread rebalanceThread;
+    private final TaskDef taskDef;
     private final LHLivenessController livenessController;
-
-    private static final int TOTAL_RETRIES = 5;
+    private static final long HEARTBEAT_INTERVAL_MS = 5000L;
```

Comment:
While we are already rewriting everything—can we move this up to like 25 or 30 seconds?